packages/braintree-integration/src/braintree-visa-checkout/braintree-visa-checkout-customer-strategy.ts

```diff
@@ -0,0 +1,253 @@
+            const state = this.paymentIntegrationService.getState();
+
+            const shipping = state.getShippingAddress();
+            const billing = state.getBillingAddress();
+
+            const {
+                shippingAddress = this.toVisaCheckoutAddress(shipping),
+                billingAddress = this.toVisaCheckoutAddress(billing),
+            } = payload;
+
+            return this.postForm(
+                {
+                    ...payload,
+                    shippingAddress,
+                    billingAddress,
+                },
+                deviceData,
+            );
+        });
+    }
+
+    private postForm(
+        paymentData: VisaCheckoutTokenizedPayload,
+        dataCollector: BraintreeDataCollector,
+    ) {
+        const { userData, billingAddress, shippingAddress, details: cardInformation } = paymentData;
+        const { userEmail } = userData;
+        const { deviceData } = dataCollector;
+
+        return this.formPoster.postForm('/checkout.php', {
+            payment_type: paymentData.type,
+            nonce: paymentData.nonce,
+            provider: 'braintreevisacheckout',
+            action: 'set_external_checkout',
+            device_data: deviceData,
+            card_information: JSON.stringify({
+                type: cardInformation.cardType,
+                number: cardInformation.lastTwo,
+            }),
+            billing_address: JSON.stringify(this.getAddress(userEmail, billingAddress)),
+            shipping_address: JSON.stringify(this.getAddress(userEmail, shippingAddress)),
+        });
+    }
+
+    private toVisaCheckoutAddress(address?: Address): VisaCheckoutAddress {
+        if (!address) {
+            return {};
+        }
+
+        return {
+            firstName: address.firstName,
+            lastName: address.lastName,
+            phoneNumber: address.phone,
+            streetAddress: address.address1,
+            extendedAddress: address.address2,
+            locality: address.city,
+            region: address.stateOrProvinceCode,
+            countryCode: address.countryCode,
+            postalCode: address.postalCode,
+        };
+    }
+
+    private getAddress(email: string, address: VisaCheckoutAddress = {}): Partial<LegacyAddress> {
+        return {
+            email,
+            first_name: address.firstName,
+            last_name: address.lastName,
+            phone_number: address.phoneNumber,
+            address_line_1: address.streetAddress,
+            address_line_2: address.extendedAddress,
+            city: address.locality,
+            state: address.region,
+            country_code: address.countryCode,
+            postal_code: address.postalCode,
+        };
+    }
+
+    private createSignInButton(containerId: string, buttonClass: string): HTMLElement {
+        const container = document.querySelector(`#${containerId}`);
+
+        if (!container) {
+            throw new Error('Need a container to place the button');
```

Comment:
```suggestion
            throw new Error('Unable to proceed because the provided container ID is not valid.');
```

I took this message from hosted fields core implementation